db/db_impl/db_impl_open.cc

```diff
@@ -319,6 +320,7 @@ Status DBImpl::NewDB(std::vector<std::string>* new_filenames) {
     }
     std::unique_ptr<FSWritableFile> file;
     FileOptions file_options = fs_->OptimizeForManifestWrite(file_options_);
+    file_options.temperature = immutable_db_options_.metadata_write_temperature;
```

Comment:
Perhaps I should not overwrite the result of `OptimizeForManifestWrite` if `metadata_write_temperature == kUnknown` (and similar below for `OptimizeForLogWrite`)

Reply:
Agreed, rest of the change is great! Thank you for adding this feature, maybe add a history entry too?